internal/datacoord/session/datanode_manager.go

```diff
@@ -507,28 +510,53 @@ func (c *DataNodeManagerImpl) DropImport(nodeID int64, in *datapb.DropImportRequ
 	return merr.CheckRPCCall(status, err)
 }
 
-func (c *DataNodeManagerImpl) CheckHealth(ctx context.Context) error {
-	group, ctx := errgroup.WithContext(ctx)
-
+func (c *DataNodeManagerImpl) CheckDNHealth(ctx context.Context) healthcheck.Result {
+	result := healthcheck.NewResult()
+	wg := sync.WaitGroup{}
+	wlock := sync.Mutex{}
 	ids := c.GetSessionIDs()
+
 	for _, nodeID := range ids {
 		nodeID := nodeID
-		group.Go(func() error {
-			cli, err := c.getClient(ctx, nodeID)
+		wg.Add(1)
+		go func() {
+			var datanodeClient types.DataNodeClient
+			var checkHealthResp *milvuspb.CheckHealthResponse
+			var err error
+
+			defer func() {
+				if err != nil {
+					wlock.Lock()
+					result.AppendUnhealthyNodeMsg(healthcheck.UnhealthyNodeMsg{
+						Role:         "DataNode",
+						NodeID:       nodeID,
+						UnhealthyMsg: err.Error(),
+					})
+					wlock.Unlock()
+				}
+				wg.Done()
+			}()
+
+			datanodeClient, err = c.getClient(ctx, nodeID)
 			if err != nil {
-				return fmt.Errorf("failed to get DataNode %d: %v", nodeID, err)
+				err = fmt.Errorf("failed to get node:%d: %v", nodeID, err)
+				return
 			}
 
-			sta, err := cli.GetComponentStates(ctx, &milvuspb.GetComponentStatesRequest{})
-			if err != nil {
-				return err
+			checkHealthResp, err = datanodeClient.CheckHealth(ctx, &milvuspb.CheckHealthRequest{})
```

Comment:
`CheckHealth` ia a new API here, for compatible reason, should consider of `not implemented` error